### PR TITLE
Added external functions for persisting Grant Tables

### PR DESCRIPTION
**Pull request by Hydrocharged**
This adds the functions necessary to be able to easily persist the data from Grant Tables as JSON for an integrator.

### CODE REVIEW

File: sql/grant_tables/privilege_set_json.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+type privilegeSetMarshalerColumn struct {
+	Name       string
+	Privileges []string
+}
+
+var _ json.Marshaler = PrivilegeSet{}
+var _ json.Unmarshaler = (*PrivilegeSet)(nil)
+
+// MarshalJSON implements the interface json.Marshaler.
+func (ps PrivilegeSet) MarshalJSON() ([]byte, error) {
```
Comment (andrew-wm-arthur) [inline sql/grant_tables/privilege_set_json.go:54]: `MarshallJSON` and `UnmarshallJSON` could both use some refactoring to split out these nested loops into helper functions

File: sql/grant_tables/grant_tables.go [hunk 2]
```diff
@@ -255,8 +290,40 @@ func (g *GrantTables) Negotiate(c *mysql.Conn, user string, addr net.Addr) (mysq
...
+		}
+		return users[i].Host < users[j].Host
+	})
+
+	roleEntries := g.role_edges.data.ToSlice(ctx)
+	roles := make([]*RoleEdge, len(roleEntries))
+	for i, roleEntry := range roleEntries {
+		roles[i] = roleEntry.(*RoleEdge)
+	}
+	sort.Slice(roles, func(i, j int) bool {
```
Comment (andrew-wm-arthur) [inline sql/grant_tables/grant_tables.go:314]: the ordering semantics here could use a comment or a helper function to clarify things

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM, just a bit of code-cleanup you can address later

File: sql/grant_tables/privilege_set_json.go (hunk 1 above)
Comment (Hydrocharged) [reply sql/grant_tables/privilege_set_json.go:54]: I'll definitely do this...in a later PR lol

File: sql/grant_tables/grant_tables.go (hunk 2 above)
Comment (Hydrocharged) [reply sql/grant_tables/grant_tables.go:314]: I'll leave a comment

**Pull request merged**